bentoml/_internal/utils/__init__.py

```diff
@@ -5,10 +5,16 @@
 import socket
 import tarfile
 from io import StringIO
-from typing import Callable, Optional
+from typing import (
+    Optional, TypeVar, Type, Union, overload, Dict, Iterator, Any, Tuple,
+    TYPE_CHECKING, Generic, Callable)
 from urllib.parse import urlparse, uses_netloc, uses_params, uses_relative
 
 from google.protobuf.message import Message
+from mypy.typeshed.stdlib.contextlib import _GeneratorContextManager
```

Comment:
This means adding mypy into our library dependency. I would rather to reserve all `typeshed` import under `TYPE_CHECKING`

Reply:
Sorry, I forgot to add the relevant dependencies.

Reply:
no worries since its only a type call i just added to the files. I think we shouldn't add mypy to library dependency anw

Reply:
Because `mypy` uses `typeshed` types to annotate the standard libraries types during static analysis, so that `mypy` can only recognize the types in `typeshed`.

![image](https://user-images.githubusercontent.com/1206493/127383421-0e10b3e7-4294-49c9-b49b-310bbc30d8bf.png)

Reply:
I see. I don't know if its a good practice to include mypy into library dependency. I don't see a lot of other library doing so.

Reply:
I'm not sure how to do it, I'll research how other libraries are doing it.